services/QuillLMS/app/controllers/cms/school_admins_controller.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class Cms::SchoolAdminsController < Cms::CmsController
+  before_action :set_school, only: [:create]
+
+  def create
+    user = User.find_by(email: params[:email])
+    if user
+      create_admin_user_for_existing_user(user)
+    else
+      create_new_account_for_admin_user
+    end
+  end
+
+  def set_school
+    @school = School.find(params[:school_id])
+  end
+
+  private def handle_admin_save(user, school_id, new_user)
+    user.mailer_user.send_school_admin_email(school_id, new_user)
+    returned_message = new_user ? t('admin.make_admin') : t('admin_created_account.existing_account.admin.new')
+
+    if params[:is_make_admin_button]
+      flash[:success] = t('admin.make_admin')
+      redirect_to cms_school_path(school_id)
+    else
+      render json: { message: returned_message }, status: 200
+    end
+  end
+
+  private def create_admin_user_for_existing_user(user, new_user: false)
+    school_id = params[:school_id]
+
+    if user.admin?
+      school_name = SchoolsAdmins.find_by(school_id: school_id).school.name
+      return render json: { message: t('admin_created_account.existing_account.admin.linked', school_name: school_name) }
+    end
+
+    admin = SchoolsAdmins.new
```

Comment:
39-41 can be rewritten: `admin = user.schools_admins.build(school_id: school_id)`

However, I would name the variable `school_admin` to better reflect its type, otherwise, `admin` feels like an object more like `user` when in fact it's a join record between a user and school

```
    school_admin = user.schools_admins.build(school_id: school_id)

    if school_admin.save!
      handle_school_admin_save(user, school_id, new_user)
    else
      render json: { error: school_admin.errors.messages }
    end
```

Reply:
that's a good point, I'll make this update